subsys/sm_ipt/CMakeLists.txt

```
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+zephyr_include_directories(include)
+
+zephyr_library()
+
+zephyr_library_sources_ifdef(CONFIG_SM_IPT sm_ipt_os.c)
```

Maybe just:
```suggestion
zephyr_library_sources(sm_ipt_os.c)
```

Add conditionally add sm_ipt directory `CONFIG_SM_IPT`